model_api/cpp/models/src/classification_model.cpp

```diff
@@ -87,16 +134,31 @@ ClassificationModel::ClassificationModel(std::shared_ptr<ov::Model>& model, cons
     } else {
         topk = topk_iter->second.as<size_t>();
     }
-    auto multilabel_iter = configuration.find("multilabel");
-    if (multilabel_iter == configuration.end()) {
-        if (model->has_rt_info("model_info", "multilabel")) {
-            std::string val = model->get_rt_info<std::string>("model_info", "multilabel");
-            multilabel = val == "True" || val == "YES";
+
+    auto thresh_iter = configuration.find("confidence_threshold");
+    if (thresh_iter == configuration.end()) {
+        if (model->has_rt_info("model_info", "confidence_threshold")) {
+            confidence_threshold = stof(model->get_rt_info<std::string>("model_info", "confidence_threshold"));
         }
     } else {
-        std::string val = multilabel_iter->second.as<std::string>();
-        multilabel = val == "True" || val == "YES";
+        confidence_threshold = thresh_iter->second.as<float>();
+    }
+
+    multilabel = get_bool_config_value("multilabel", model, configuration);
+    hierarchical = get_bool_config_value("hierarchical", model, configuration);
+
+    auto config_iter = configuration.find("hierarchical_config");
+    if (config_iter == configuration.end()) {
+        if (model->has_rt_info("model_info", "hierarchical_config")) {
+            hierarchical_json_config = model->get_rt_info<std::string>("model_info", "hierarchical_config");
+        }
+    } else {
+        hierarchical_json_config = thresh_iter->second.as<std::string>();
     }
+    if (hierarchical_json_config.size())  {
```

Comment:
You should try to read it only `if hierarchical`. But `if (hierarchical && hierarchical_json_config.empty())` this should be an error.

Reply:
Done